src/feedClass.ts

```diff
@@ -0,0 +1,281 @@
+                    result[keyItem] = _.pick(mainFields, this.feedItemFields);
+                    const dataItem = await this.feeds
+                        .get(feedInfoName)
+                        .get('items')
+                        .get(keyItem)
+                        .map()
+                        .get('data')
+                    result[keyItem].data = _.pick(dataItem, this.feedItemDataFields)
+                }
+            }
+        }
+        return result
+    }
+
+    /**
+     * Return array items
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeedPrivateItems(feedInfoName: string): Promise<FeedItem | any> {
+        let result: any = {};
+        logger.info('class getFeedPrivateItems', {feedInfoName});
+        const privateItems = await this.feeds.get(feedInfoName).get('privateItems').then();
+        if (privateItems) {
+            for (let privateKey in privateItems) {
+                if (privateKey !== '_') {
+                    result[privateKey] = {};
+                    let items = await this.feeds
+                        .get(feedInfoName)
+                        .get('privateItems')
+                        .get(privateKey)
+                        .then();
+                    if (items) {
+                        for (let keyItem in items) {
+                            if (keyItem !== '_') {
+                                const mainFields = await this.feeds
+                                    .get(feedInfoName)
+                                    .get('privateItems')
+                                    .get(privateKey)
+                                    .get(keyItem)
+                                    .map()
+                                console.log('**********************')
+                                console.log(mainFields)
+                                console.log('**********************')
+                                result[privateKey][keyItem] = _.pick(mainFields, this.feedItemFields);
+                                const dataItem = await this.feeds
+                                    .get(feedInfoName)
+                                    .get('privateItems')
+                                    .get(privateKey)
+                                    .get(keyItem)
+                                    .map()
+                                    .get('data')
+                                result[privateKey][keyItem].data = _.pick(dataItem, this.feedItemDataFields)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return result
+    }
+
+    /**
+     * Get all feeds user's
+     */
+    async getFeeds(): Promise<any> {
+        logger.info('class getFeeds');
+        let result = {};
+        try {
+            const feeds = await this.feeds.then();
+
+            if (feeds) {
+                for (let feeedKey in feeds) {
+                    if (feeedKey !== '_') {
```

Comment:
filtering for `key !== '_'` performs in a few places. It could be moved somewere
in the utils for example in the lib/gunnb

```
export async function fetchItemsRef(ref: Gun): Promise<Array<any>> {
  const allItems = await ref.then();

  return allItems.filter((item: any) => item !== '_');
}
```